### PR TITLE
Add FormEntity (trello [009][010])

### PR DESCRIPTION
**Pull request by ShirshovAlex**
add Form Entity 
add Form Repository
add Form Service
add Form mapstruct
change Form endpoint (working with DB)

### CODE REVIEW

**Review by OlgaEvdokimova — DISMISSED** [review]

File: src/main/java/by/exadel/internship/repository/FormRepository.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import by.exadel.internship.entity.Form;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import javax.transaction.Transactional;
+import java.util.UUID;
+
+@Repository
+@Transactional
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/repository/FormRepository.java]: Does @Transactional make sense here?

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/repository/FormRepository.java]: remove it

**Review by OlgaEvdokimova — DISMISSED** [review]

File: src/main/java/by/exadel/internship/mapper/FormMapper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+public interface FormMapper {
+
+    FormFullDTO toFormDto(Form form);
+
+    Form toFormEntity(FormFullDTO formDTO);
+
+    FormRegisterDTO toFormRegDTO(Form form);
+
+    Form toFormEntity(FormRegisterDTO formRegisterDTO);
+
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/mapper/FormMapper.java]: Remove excessive empty lines, please.

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java [hunk 3]
```diff
@@ -6,13 +6,14 @@
...
 
+
 @Data
 @SuperBuilder
 @NoArgsConstructor
 @AllArgsConstructor
 public class FormRegisterDTO {
 
-    private String name;
+    private String firstName;
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java:16]: Add please validation annotations for required fields.

Comment (kzaretskaya) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java:16]: +  and regex for email

File: src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package by.exadel.internship.team_four;
```
Comment (kzaretskaya) [inline src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java]: there's better to store test to the same test package as testing class.
```suggestion
package by.exadel.internship.mapper;
```

File: src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public void givenDTOtoEntity_whenMaps_thenCorrect() {
+        FormFullDTO formFullDTO = new FormFullDTO();
+        formFullDTO.setEmail("testEmail");
+        formFullDTO.setEducation("testEducation");
+        Form form = mapper.toFormEntity(formFullDTO);
+
+        assertEquals(formFullDTO.getEducation(), form.getEducation());
+        assertEquals(formFullDTO.getEmail(), form.getEmail());
+    }
+
```
Comment (kzaretskaya) [inline src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java]: empty lines

Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java]: done

File: src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java (hunk 4 above)
Comment (ShirshovAlex) [reply src/test/java/by/exadel/internship/team_four/SimpleFormMapperTest.java]: done

File: src/main/java/by/exadel/internship/mapper/FormMapper.java (hunk 2 above)
Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/mapper/FormMapper.java]: done

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java (hunk 3 above)
Comment (mortal-gpt) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java:16]: @kzaretskaya @ShirshovAlex  The validation Library has special annotation @Email, so there is no need to use regular expressions.

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java:16]: we'll do it later

**Review by OlgaEvdokimova — APPROVED** [review]

**Review by kzaretskaya — APPROVED** [review]

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java (hunk 3 above)
Comment (kzaretskaya) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java:16]: Create, please, in Trello appropriate ticket for keeping in mind this.

File: src/main/java/by/exadel/internship/controller/FormController.java [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
+    @Autowired
+    public FormService formService;
 
     @PostMapping
     @ApiOperation("Add new form")
-    public FormRegisterDTO addNewForm(@RequestBody FormRegisterDTO form) {
-        formList.add(form);
-        return form;
+    public void addNewForm(@RequestBody FormRegisterDTO form) {
+        formService.saveForm(form);
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/controller/FormController.java:21]: Should this endpoint return ResponseEntity?

Comment (ShirshovAlex) [reply src/main/java/by/exadel/internship/controller/FormController.java:21]: il do it in next commit

**Pull request merged**